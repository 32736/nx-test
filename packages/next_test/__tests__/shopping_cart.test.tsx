import { render } from '@testing-library/react';

import ShoppingCart from '@/pages/shopping_cart';

describe('ShoppingCart', () => {
  it('render products', () => {
    const products = [
        { name: 'prod 1', price: 100 },
        { name: 'prod 2', price: 200 },
        { name: 'prod 3', price: 300 }
    ];

    const { container } = render(<ShoppingCart products={products} />);

    expect(container).toMatchSnapshot();
  });
});
