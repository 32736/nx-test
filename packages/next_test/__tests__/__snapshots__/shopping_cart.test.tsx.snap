// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingCart render products 1`] = `
<div>
  <div
    class="container"
  >
    <main
      class="main"
    >
      <h1
        class="title"
      >
        shopping_cart
      </h1>
      <table
        cellspacing="0"
        class="table"
      >
        <colgroup>
          <col
            width="200"
          />
          <col
            width="200"
          />
          <col
            width="200"
          />
          <col
            width="200"
          />
        </colgroup>
        <thead>
          <tr>
            <th>
              name
            </th>
            <th>
              price
            </th>
            <th>
              count
            </th>
            <th />
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              prod 1
            </td>
            <td>
              100
            </td>
            <td>
              0
            </td>
            <td>
              <button
                class="button"
              >
                add
              </button>
            </td>
          </tr>
          <tr>
            <td>
              prod 2
            </td>
            <td>
              200
            </td>
            <td>
              0
            </td>
            <td>
              <button
                class="button"
              >
                add
              </button>
            </td>
          </tr>
          <tr>
            <td>
              prod 3
            </td>
            <td>
              300
            </td>
            <td>
              0
            </td>
            <td>
              <button
                class="button"
              >
                add
              </button>
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colspan="4"
            >
              total: 
              0
            </td>
          </tr>
        </tfoot>
      </table>
    </main>
  </div>
</div>
`;
